### ISSUE TITLE
Cilium pods with well-known identities get stuck in "restoring" state if etcd is down

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#20166]
Saw Cilium restore logic get stuck while restoring a pod with a well-known identity:
```
github.com/cilium/cilium/pkg/ipcache.(*IPIdentityWatcher).waitForInitialSync(...)
        /go/src/github.com/cilium/cilium/pkg/ipcache/kvstore.go:388
github.com/cilium/cilium/pkg/ipcache.WaitForKVStoreSync(...)
        /go/src/github.com/cilium/cilium/pkg/ipcache/kvstore.go:412
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).restoreIdentity(0xc00044d500, 0xc0004deca0, 0x4432ca)
        /go/src/github.com/cilium/cilium/pkg/endpoint/restore.go:333 +0x9cf
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).RegenerateAfterRestore(0xc00044d500, 0x0, 0x0)
        /go/src/github.com/cilium/cilium/pkg/endpoint/restore.go:230 +0x45
github.com/cilium/cilium/daemon/cmd.(*Daemon).regenerateRestoredEndpoints.func1(0xc00044d500, 0xc000af8f50)
        /go/src/github.com/cilium/cilium/daemon/cmd/state.go:292 +0x2b
created by github.com/cilium/cilium/daemon/cmd.(*Daemon).regenerateRestoredEndpoints
        /go/src/github.com/cilium/cilium/daemon/cmd/state.go:291 +0x6d4
```

Restoring such a pod should not wait on etcd to resolve the identity, because the identity is well-known.

**Comment by joestringer** [comment]
Won't fix.
